pyracf/common/logger.py

```diff
@@ -48,24 +49,25 @@ def log_dictionary(
         self,
         header_message: str,
         dictionary: dict,
-        redact_strings: List[str] = [],
+        secret_traits: dict = {},
     ) -> None:
         """JSONify and colorize a dictionary and log it to the console."""
+        if secret_traits:
+            dictionary = self.__redact_request_dictionary(dictionary, secret_traits)
         dictionary_json = json.dumps(dictionary, indent=2)
-        dictionary_json = self.redact_strings(dictionary_json, redact_strings)
         colorized_dictionary_json = self.__colorize_json(dictionary_json)
         self.log_debug(header_message, colorized_dictionary_json)
 
     def log_xml(
         self,
         header_message: str,
         xml_string: Union[str, bytes],
-        redact_strings: List[str] = [],
+        secret_traits: dict = {},
     ) -> None:
         """Indent and colorize XML string and log it to the console."""
         if isinstance(xml_string, bytes):
             xml_string = xml_string.decode(encoding="utf-8")
-        xml_string = self.redact_strings(xml_string, redact_strings)
+        xml_string = self.redact_request_xml(xml_string, secret_traits)
```

Comment:
I think we can do something like this similar to `log_dictionary` here.

```python
        if secret_traits:
            xml_string = self.redact_request_xml(xml_string, secret_traits)
```

Reply:
Agreed. Done.